nixos/modules/programs/looking-glass/default.nix

```diff
@@ -0,0 +1,188 @@
+{config, pkgs, lib, ...}:
+let
+  cfg = config.programs.looking-glass;
+
+  # Generate the looking-glass-client configuration where boolean values
+  # are encoded as "yes" or "no".
+  generateConfig = settings: lib.generators.toINI {
+    mkKeyValue = lib.generators.mkKeyValueDefault {
+      mkValueString = v:
+        if (true == v) then "yes"
+        else if (false == v) then "no"
+        else (lib.generators.mkValueStringDefault {} v);
+    } "=";
+  } settings;
```

Comment:
It is not recommended to add all config options of a specific program into its NixOS module due to issues such as maintenance burden.  For more info, [rfc 42][rfc] is a good read.

Only some important settings should be defined in the NixOS module.  This [doc] guides you how to do so and here is an [example][example].

[rfc]: https://github.com/NixOS/rfcs/blob/master/rfcs/0042-config-option.md
[doc]: https://nixos.org/manual/nixos/unstable/#sec-settings-options
[example]: https://nixos.org/manual/nixos/unstable/#sec-settings-attrs-options

Reply:
I appreciate the feedback! I'll take a look at that documentation, and revisit.